From c863f85bd24119660f3b4af554a582926db59310 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Thu, 12 Apr 2012 16:24:22 +0800
Subject: [PATCH 261/502] ENGR00179512 MX6 libgralloc:Correct display refresh
 calc

The code in framebuffer.cpp doesn't get display refresh
rate correctly, as it doen't consider vsync_length and
hsync_length. This patch adds info.vsync_len and
info.hsync_len to the calculation formula.
It helps to pass display refresh rate cts.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
---
 mx6/libgralloc/framebuffer.cpp         | 4 ++--
 mx6/libgralloc_wrapper/framebuffer.cpp | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/mx6/libgralloc/framebuffer.cpp b/mx6/libgralloc/framebuffer.cpp
index 37ab14e..24c5461 100755
--- a/mx6/libgralloc/framebuffer.cpp
+++ b/mx6/libgralloc/framebuffer.cpp
@@ -954,8 +954,8 @@ int mapFrameBufferLocked(struct private_module_t* module)
 #endif
     int refreshRate = 1000000000000000LLU /
     (
-            uint64_t( info.upper_margin + info.lower_margin + info.yres )
-            * ( info.left_margin  + info.right_margin + info.xres )
+            uint64_t(info.upper_margin + info.lower_margin + info.yres + info.vsync_len)
+            * (info.left_margin  + info.right_margin + info.xres + info.hsync_len)
             * info.pixclock
     );
 
diff --git a/mx6/libgralloc_wrapper/framebuffer.cpp b/mx6/libgralloc_wrapper/framebuffer.cpp
index 01e20fb..0ecc340 100755
--- a/mx6/libgralloc_wrapper/framebuffer.cpp
+++ b/mx6/libgralloc_wrapper/framebuffer.cpp
@@ -425,8 +425,8 @@ static int mapFrameBufferWithParamLocked(struct private_module_t* module, struct
 
     int refreshRate = 1000000000000000LLU /
     (
-            uint64_t( info.upper_margin + info.lower_margin + info.yres )
-            * ( info.left_margin  + info.right_margin + info.xres )
+            uint64_t(info.upper_margin + info.lower_margin + info.yres + info.vsync_len)
+            * (info.left_margin  + info.right_margin + info.xres + info.hsync_len)
             * info.pixclock
     );
 
-- 
1.8.0

